src/common/imageio_dng.h

```diff
@@ -97,34 +95,43 @@ static inline void dt_imageio_dng_write_tiff_header(
   buf[3] = 42;
   buf[7] = 10;
 
-  b = buf + 12;
-  b = dt_imageio_dng_make_tag(254, LONG, 1, 0, b, &cnt);           /* New subfile type.  */
-  b = dt_imageio_dng_make_tag(256, SHORT, 1, (xs << 16), b, &cnt); /* Image width.  */
-  b = dt_imageio_dng_make_tag(257, SHORT, 1, (ys << 16), b, &cnt); /* Image length.  */
-  // b = dt_imageio_dng_make_tag(  258, SHORT, channels, 506, b, &cnt ); /* Bits per sample.  */
-  b = dt_imageio_dng_make_tag(258, SHORT, 1, 32 << 16, b, &cnt); /* Bits per sample.  */
-  // bits per sample: 32-bit float
-  // buf[507] = buf[509] = buf[511] = 32;
-  b = dt_imageio_dng_make_tag(259, SHORT, 1, (1 << 16), b, &cnt); /* Compression.  */
-  b = dt_imageio_dng_make_tag(262, SHORT, 1, 32803 << 16, b, &cnt);
-      /* cfa */ // 34892, b, &cnt ); // linear raw /* Photo interp.  */
-  // b = dt_imageio_dng_make_tag(  271, ASCII, 8, 494, b, &cnt); // maker, needed for dcraw
-  // b = dt_imageio_dng_make_tag(  272, ASCII, 9, 484, b, &cnt); // model
-  //   offs2 = b + 8;
-  b = dt_imageio_dng_make_tag(273, LONG, 1, 584, b, &cnt);             /* Strip offset.  */
-  b = dt_imageio_dng_make_tag(274, SHORT, 1, 1 << 16, b, &cnt);        /* Orientation. */
-  b = dt_imageio_dng_make_tag(277, SHORT, 1, channels << 16, b, &cnt); /* Samples per pixel.  */
-  b = dt_imageio_dng_make_tag(278, SHORT, 1, (ys << 16), b, &cnt);     /* Rows per strip.  */
-  b = dt_imageio_dng_make_tag(279, LONG, 1, (ys * xs * channels * 4), b,
-                              &cnt);                              // 32 bits/channel /* Strip byte count.  */
-  b = dt_imageio_dng_make_tag(284, SHORT, 1, (1 << 16), b, &cnt); /* Planar configuration.  */
-  b = dt_imageio_dng_make_tag(339, SHORT, 1, (3 << 16), b,
-                              &cnt); /* SampleFormat = 3 => ieee floating point */
+  uint8_t *b = buf + 12;
+  int data = 512; // take care of room in data segment; the tags may use up to 500 bytes so far 
```

Comment:
Btw, maybe we can do even better - as we know how many tags we'll be writing, we should know where the data will start precisely so there are no gaps... I can maybe take a look at this sometime later?

Reply:
We know **now** how many tags we are writing so no problem to decrease this gap's size but ... we might want to define more and we would have to recalculate again and again.

Reply:
Sure, but we can just have a formula: 8 + (2 + Ntags * 12) + 2, no?

Having a fixed "512" (or whatever value) runs a risk that if/when you add more tags it will start overlapping at some point so you'd need to figure it out and change it anyway...

Reply:
Since you're getting rid of magic numbers, it makes sense to do the same for tag IDs. Either enum or #defines can be used for that, and then you could do some preprocessor magic using __LINE__ to get the number of used (well, defined) tags. Adding another tag will be handled automatically in this case, as long as code review guarantees magic numbers aren't used as tag IDs anymore.

Reply:
> then you could do some preprocessor magic using `__LINE__` to get the number

Aah, thanks - i have never been using this macro before.

Reply:
done